SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -169,30 +174,78 @@ public override bool InstanceEnabled
+        }
+
+        private Task<ErrorReport> InstanceGetLastSessionCrashReportAsync()
+        {
+            return _lastSessionErrorReportTaskSource?.Task ?? Task.FromResult<ErrorReport>(null);
+        }
+
         private Task ProcessPendingErrorsAsync()
         {
             return Task.Run(async () =>
             {
-                foreach (var file in ErrorLogHelper.GetErrorLogFiles())
+                var lastSessionErrorLogTimestamp = DateTime.MinValue;
+                ManagedErrorLog lastSessionErrorLog = null;
+                try
                 {
-                    AppCenterLog.Debug(LogTag, $"Process pending error file {file.Name}");
-                    var log = ErrorLogHelper.ReadErrorLogFile(file);
-                    if (log == null)
+                    foreach (var file in ErrorLogHelper.GetErrorLogFiles())
                     {
-                        AppCenterLog.Error(LogTag, $"Error parsing error log. Deleting invalid file: {file.Name}");
+                        AppCenterLog.Debug(LogTag, $"Process pending error file {file.Name}");
+                        var log = ErrorLogHelper.ReadErrorLogFile(file);
+
+                        // Process the file for last session crash report. It doesn't matter if the log is null.
                         try
                         {
-                            file.Delete();
+                            var otherFileTimestamp = file.LastWriteTime;
+                            if (lastSessionErrorLogTimestamp < otherFileTimestamp)
+                            {
+                                lastSessionErrorLogTimestamp = otherFileTimestamp;
+                                lastSessionErrorLog = log;
+                            }
                         }
                         catch (System.Exception ex)
                         {
-                            AppCenterLog.Warn(LogTag, $"Failed to delete error log file {file.Name}.", ex);
+                            AppCenterLog.Warn(LogTag, $"Failed to get the last write time for an error file.", ex);
+                        }
+
+                        // Finish processing the file.
+                        if (log == null)
+                        {
+                            AppCenterLog.Error(LogTag, $"Error parsing error log. Deleting invalid file: {file.Name}");
+                            try
+                            {
+                                file.Delete();
+                            }
+                            catch (System.Exception ex)
+                            {
+                                AppCenterLog.Warn(LogTag, $"Failed to delete error log file {file.Name}.", ex);
+                            }
+                        }
+                        else
+                        {
+                            _unprocessedManagedErrorLogs.Add(log.Id, log);
                         }
                     }
+                }
+                finally
```

Comment:
why is that in a finally? if there is an unhandled exception here it will crash the app

Reply:
what if it doesn't crash the app? This could be running on any thread. Or, what if there's just a missed branch? I guess it's a little defensive but I'm paranoid about the code skipping actually completing the task.

Reply:
We are usually not defensive as code needs to be unit tested.
If there is a missed branch please review the code again.